private/tools/java/com/github/bazelbuild/rules_jvm_external/maven/MavenPublisher.java

```diff
@@ -174,10 +203,27 @@ private static CompletableFuture<Void> upload(
     uploads.add(upload(String.format("%s%s.md5", base, append), credentials, md5));
     uploads.add(upload(String.format("%s%s.sha1", base, append), credentials, sha1));
 
-    if (gpgSign) {
-      uploads.add(upload(String.format("%s%s.asc", base, append), credentials, sign(item)));
-      uploads.add(upload(String.format("%s%s.md5.asc", base, append), credentials, sign(md5)));
-      uploads.add(upload(String.format("%s%s.sha1.asc", base, append), credentials, sign(sha1)));
+    SigningMethod signingMethod = signingMetadata.getSigningMethod();
```

Comment:
Given how this class is used, it feels like it might be an interface with three implementations (in-memory, gpg, and "not signing") and two methods ("sign" and "isSigning"). That might make this a little clearer

Reply:
This should be doable. However, seems to be a bit overkill due to the following reasons:
1. The change scope will be much larger than the issue this PR was originally planning to address and needs a more thorough test plan.
1. There's no dependency injection support in this module, and most of the signing logic are defined inside static method, which diminish the benefits of defining interface/implementation.
1. Technically speaking, there's only 2 sub-types (gpg, pgp) and most likely we won't add any new signing support in the future. Make not-signing to apply `sign` method along with 2 other use cases also seems counter-intuitive.